subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -59,7 +60,7 @@ int lwm2m_firmware_start_transfer(char *package_uri)
 
 	/* start file transfer work */
 	strncpy(fota_context.uri, package_uri, LWM2M_PACKAGE_URI_LEN - 1);
-	lwm2m_pull_context_start_transfer(&fota_context);
+	lwm2m_pull_context_start_transfer(&fota_context, K_NO_WAIT);
```

Comment:
This is basically what makes it potentially controversial. What happens if the sem is taken here? Is the choice K_NO_WAIT clear cut?

Reply:
It is not.

But IMO the choices are K_NO_WAIT or K_FOREVER. I choose K_NO_WAIT in order to not keep the lwm2m_rx thread

Reply:
Maybe there should be a q

Reply:
If the sem is taken, the FOTA system notifies the cloud that the FOTA failed at least right? So it can be retriggered?

Reply:
Yes the write on the uri resource will "fail". This  is a bit missleading but there is nothing better we can do as writing to the uri is the trigger for starting the download and setting the uri as well.